nodes/src/nodes/cost_tracker/IInstance.py

```
@@ -0,0 +1,110 @@
+        budget_status = tracker.check_budget()
+
+        if budget_status.get('alert_threshold_reached') and budget_status.get('within_budget'):
+            debug(f'Cost Tracker: alert threshold reached -- {budget_status["percent_used"]:.1f}% of budget used (${budget_status["used"]:.6f} / ${budget_status["used"] + budget_status["remaining"]:.6f})')
+
+        if not budget_status.get('within_budget', True):
+            if tracker.policy == 'block':
+                debug(f'Cost Tracker: BLOCKING answer -- budget exceeded (${budget_status["used"]:.6f} spent, limit ${budget_status["used"] - budget_status["remaining"]:.6f})')
+                # Prevent the answer from propagating downstream
+                self.preventDefault()
+                return
+            else:
```

Ruff (0.15.7)

[warning] Unnecessary `else` after `return` statement

Remove unnecessary `else`

(RET505)